fails to build with text-2.1.2

For Stackage Nightly with ghc-9.10.2

```
       Building library 'testlib' for rest-rewrite-0.4.4...                                                    12:58:25 [485/954403]
       [1 of 7] Compiling Language.REST.ConcreteOC                                                                                  
       [2 of 7] Compiling Language.REST.ProofGen                  
       [3 of 7] Compiling MultisetOrdering
       [4 of 7] Compiling Nat                                     
       /home/curators/work/unpack-dir/unpacked/rest-rewrite-0.4.4-b09d3ab6fa4e5924f3eb712406c8812e959747f47b4621db3e3509ffaea66c44/t
estlib/Nat.hs:49:24: error: [GHC-87543]      
           Ambiguous occurrence ‘show’.                
           It could refer to                                      
              either ‘Prelude.show’,                       
                     imported from ‘Prelude’ at testlib/Nat.hs:7:8-10                                                               
                     (and originally defined in ‘ghc-internal-9.1002.0:GHC.Internal.Show’),                                         
                  or ‘Data.Text.show’,             
                     imported from ‘Data.Text’ at testlib/Nat.hs:10:1-16.                                                           
          |                                                       
       49 |     showInt t = pack . show <$> termToInt t
          |                        ^^^^              
                                                                  
       /home/curators/work/unpack-dir/unpacked/rest-rewrite-0.4.4-b09d3ab6fa4e5924f3eb712406c8812e959747f47b4621db3e3509ffaea66c44/t
estlib/Nat.hs:99:24: error: [GHC-87543]     
           Ambiguous occurrence ‘show’.              
           It could refer to                                      
              either ‘Prelude.show’,              
                     imported from ‘Prelude’ at testlib/Nat.hs:7:8-10                                                               
                     (and originally defined in ‘ghc-internal-9.1002.0:GHC.Internal.Show’),                                         
                  or ‘Data.Text.show’,         
                     imported from ‘Data.Text’ at testlib/Nat.hs:10:1-16.                                                           
          |                                                                                                                         
       99 |     Left err -> error (show err)          
          |                        ^^^^ 
```

--- a/.github/workflows/workflow.yaml
+++ b/.github/workflows/workflow.yaml
@@ -16,7 +16,7 @@ jobs:
         with:
           nix_path: nixpkgs=./nixpkgs.nix
       - name: Mount dependency cache
-        uses: actions/cache@v2
+        uses: actions/cache@v4
         with:
           path: |
             ~/.stack

--- a/rest-rewrite.cabal
+++ b/rest-rewrite.cabal
@@ -61,7 +61,7 @@ library
                 , parsec               >= 3.1.14 && < 3.2
                 , mtl                  >= 2.2.2 && < 2.4
                 , unordered-containers >= 0.2.13 && < 0.3
-                , text                 >= 1.2.4 && < 2.2
+                , text                 >= 1.2.4 && < 2.1.2
 
 library testlib
   default-language:  Haskell2010

--- a/rest-rewrite.cabal
+++ b/rest-rewrite.cabal
@@ -1,6 +1,6 @@
 name:               rest-rewrite
 build-type:         Simple
-version:            0.4.4
+version:            0.4.5
 cabal-version:      2.0
 category:           Rewriting
 maintainer:         Zack Grannan <zgrannan@cs.ubc.ca>

--- a/rest-rewrite.cabal
+++ b/rest-rewrite.cabal
@@ -61,7 +61,7 @@ library
                 , parsec               >= 3.1.14 && < 3.2
                 , mtl                  >= 2.2.2 && < 2.4
                 , unordered-containers >= 0.2.13 && < 0.3
-                , text                 >= 1.2.4 && < 2.1.2
+                , text                 >= 1.2.4 && < 2.2
 
 library testlib
   default-language:  Haskell2010

--- a/testlib/Nat.hs
+++ b/testlib/Nat.hs
@@ -46,7 +46,7 @@ pp = prettyPrint (PPArgs []
                   ] showInt)
   where
     showInt :: MT.MetaTerm -> Maybe Text
-    showInt t = pack . show <$> termToInt t
+    showInt t = pack . Prelude.show <$> termToInt t
 
 op :: GenParser Char st Op
 op = fmap (Op . pack) (many (alphaNum <|> char '\''))
@@ -96,7 +96,7 @@ term = try infixTerm <|> nonInfixTerm
 parseTerm :: String -> RuntimeTerm
 parseTerm str =
   case parse term "" str of
-    Left err -> error (show err)
+    Left err -> error (Prelude.show err)
     Right t  -> t
 
 instance IsString RuntimeTerm where

